### PR TITLE
[T3A4][T11-A1]Justin Wong

### PR DESCRIPTION
**Pull request by justinhw**

### CODE REVIEW

**Comment by JustinSYDE** [comment]
Ready to review.

File: src/seedu/addressbook/storage/StorageFile.java [hunk 1]
```diff
@@ -91,7 +91,7 @@ private static boolean isValidPath(Path filePath) {
      *
      * @throws StorageOperationException if there were errors converting and/or storing data to file.
      */
-    public void save(AddressBook addressBook) throws StorageOperationException {
+    public void save(AddressBook addressBook) throws StorageOperationException, FileNotFoundException {
```
Comment (nirandiw) [inline src/seedu/addressbook/storage/StorageFile.java:94]: We were expecting you to do an explicit check if the file exists and if it does not to throw an exception. 
The exception should be thrown from StorageFile::save() method.
The best place to catch and terminate is Main::executeCommand() (this method already has a catch block to terminate upon exception.

File: src/seedu/addressbook/storage/StorageFile.java [hunk 2]
```diff
@@ -116,7 +118,7 @@ public void save(AddressBook addressBook) throws StorageOperationException {
      *
      * @throws StorageOperationException if there were errors reading and/or converting data from file.
      */
-    public AddressBook load() throws StorageOperationException {
+    public AddressBook load() throws StorageOperationException, FileNotFoundException {
```
Comment (nirandiw) [inline src/seedu/addressbook/storage/StorageFile.java:121]: Purpose is to check accidental file deletes and not check missing file initially so load() does not make such sense.

**Review by nirandiw — COMMENTED** [review]
@JustinSYDE Some comments added. Please close the PR after reading comments.

**Pull request closed without merging**